NextcloudTalk/DiagnosticsTableViewController.swift

```diff
@@ -0,0 +1,654 @@
+/**
+ * @copyright Copyright (c) 2022 Marcel Müller <marcel-mueller@gmx.de>
+ *
+ * @author Marcel Müller <marcel-mueller@gmx.de>
+ *
+ * @license GNU GPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import UIKit
+import Contacts
+import Photos
+import UserNotifications
+
+enum DiagnosticsSections: Int {
+    case kDiagnosticsSectionApp = 0
+    case kDiagnosticsSectionAccount
+    case kDiagnosticsSectionServer
+    case kDiagnosticsSectionTalk
+    case kDiagnosticsSectionSignaling
+    case kDiagnosticsSectionCount
+}
+
+enum AppSections: Int {
+    case kAppSectionName = 0
+    case kAppSectionVersion
+    case kAppSectionAllowNotifications
+    case kAppSectionAllowMicrophoneAccess
+    case kAppSectionAllowCameraAccess
+    case kAppSectionAllowContactsAccess
+    case kAppSectionAllowLocationAccess
+    case kAppSectionAllowPhotoLibraryAccess
+    case kAppSectionOpenSettings
+    case kAppSectionCount
+}
+
+enum AccountSections: Int {
+    case kAccountSectionServer = 0
+    case kAccountSectionUser
+    case kAccountPushSubscribed
+    case kAccountSectionCount
+}
+
+enum ServerSections: Int {
+    case kServerSectionName = 0
+    case kServerSectionVersion
+    case kServerSectionWebDavRoot
+    case kServerSectionUserStatusSupported
+    case kServerSectionReachable
+    case kServerSectionCount
+}
+
+enum TalkSections: Int {
+    case kTalkSectionVersion = 0
+    case kTalkSectionCanCreate
+    case kTalkSectionCallEnabled
+    case kTalkSectionAttachmentsAllowed
+    case kTalkSectionCount
+}
+
+enum SignalingSections: Int {
+    case kSignalingSectionApiVersion = 0
+    case kSignalingSectionIsExternal
+    case kSignalingSectionStunServers
+    case kSignalingSectionTurnServers
+    case kSignalingSectionCount
+}
```

Comment:
We could move all this enums inside DiagnosticsTableViewController.
I think there is no need of having them as global variables.

Reply:
Yep, totally right. We should do the same in SettingsTableViewController then: https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/SettingsTableViewController.swift